spk/pyload/src/installer.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/sh
+
+# Package
+PACKAGE="pyload"
+DNAME="pyLoad"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+SSS="/var/packages/${PACKAGE}/scripts/start-stop-status"
+PYTHON_DIR="/usr/local/python"
+PATH="${INSTALL_DIR}/bin:${INSTALL_DIR}/env/bin:${PYTHON_DIR}/bin:${PATH}"
+USER="pyload"
+GROUP="users"
+VIRTUALENV="${PYTHON_DIR}/bin/virtualenv"
+CFG_FILE="${INSTALL_DIR}/etc/pyload.conf"
+TMP_DIR="${SYNOPKG_PKGDEST}/../../@tmp"
+
+SERVICETOOL="/usr/syno/bin/servicetool"
+FWPORTS="/var/packages/${PACKAGE}/scripts/${PACKAGE}.sc"
+
+lng2iso()
+{
+	# changes 3-character Synology language code to ISO 639-1 code.
+        case $1 in
+            eng|deu|fre|ita|nld|sve|rus|plk|csy)
+                echo "${1%?}"
+                ;;
+            spn)
+                echo "es"
+                ;;
+            *)
+		echo "en"
+		;;
+        esac
+}
+
+
+preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        test "${wizard_download_dir}" || wizard_download_dir="/volume1/downloads"
+
+        # Check whether ${wizard_download_dir} is in existing shared folder.
+        VOLUME=$(echo "${wizard_download_dir}" | cut -d/ -f2)
+        WIZ_SHARE=$(echo "${wizard_download_dir}" | cut -d/ -f3)
+        SHARE_EXISTS=$(synoshare --get "${WIZ_SHARE}" | sed -ne 's/^[[:space:]]*Path[[:space:]\.]*\[\(.*\)\].*$/\1/p')  # FIXME: Better variable name
+
+        if [ -z "${SHARE_EXISTS}" ]; then
+               echo "ERROR: Share '${WIZ_SHARE}' doesn't exist."
+               exit 1
+        fi
+
+        if [ ! -d ${wizard_download_dir} ]; then
+            echo "ERROR: Download directory ${wizard_download_dir} doesn't exist."
+            exit 1
+        fi
```

Comment:
Here's why: synoshare fails to recognize a 'normal' unix dir, because of the missing properties (ACLs and whatever else Synology might add to shared folders)
On the other hand, a normal dir check works on both.